crates/cargo-test-support/src/paths.rs

```diff
@@ -447,11 +447,12 @@ pub fn sysroot() -> String {
 #[cfg(windows)]
 pub fn windows_reserved_names_are_allowed() -> bool {
     use std::ffi::OsStr;
+    use std::iter::once;
     use std::os::windows::ffi::OsStrExt;
     use std::ptr;
     use windows_sys::Win32::Storage::FileSystem::GetFullPathNameW;
 
-    let test_file_name: Vec<_> = OsStr::new("aux.rs").encode_wide().collect();
+    let test_file_name: Vec<_> = OsStr::new("aux.rs").encode_wide().chain(once(0)).collect();
```

Comment:
You could save an import by using `[0]`. But either way this looks good to me
```suggestion
    let test_file_name: Vec<_> = OsStr::new("aux.rs").encode_wide().chain([0]).collect();
```